observatory/results/results.js

```diff
@@ -0,0 +1,167 @@
+import { LitElement, css, html } from "lit";
+import { Task } from "@lit/task";
+
+import "../feedback.js";
+import { Rating } from "./rating.js";
+import { Tabs } from "./tabs.js";
+
+import { OBSERVATORY_API_URL } from "../constants.js";
+
+/**
+ * @typedef {import("../constants.js").ObservatoryScanResult} ObservatoryScanResult
+ * @typedef {import('lit').PropertyDeclarations} PropertyDeclarations
+ * @typedef {import('../constants.js').GradeDistribution} GradeDistribution
+ * @typedef {import("@mdn/rari").SPAPage} SPAPage
+ */
+
+export class Results extends LitElement {
+  static styles = css`
+    :host {
+      --button-color: blue;
+    }
+  `;
+
+  constructor() {
+    super();
+    /** @type { string | null } */
+    this.host = null;
+    /** @type { number } */
+    this.selectedTab = 0;
+    /** @type { boolean } */
+    this._usePostInApi = false;
+  }
+
+  /**
+   * @type PropertyDeclarations
+   */
+  static properties = {
+    host: { type: String },
+    selectedTab: { type: Number },
+    _usePostInApi: { type: Boolean },
+  };
+
+  _apiTask = new Task(this, {
+    task: async ([host], { signal }) => {
+      if (!host) {
+        throw new Error("No host provided");
+      }
+      try {
+        const res = await fetch(
+          `${OBSERVATORY_API_URL}/api/v2/analyze?host=${encodeURIComponent(
+            host
+          )}`,
+          { signal, method: this._usePostInApi ? "POST" : "GET" }
+        );
+        if (!res.ok) {
+          let message = `${res.status}: ${res.statusText}`;
+          try {
+            const data = await res.json();
+            if (data.error) {
+              message = data.message;
+            }
+          } finally {
+            throw new Error(message);
+          }
+        }
+        return await res.json();
+      } catch (e) {
+        throw new Error("Observatory API request for scan data failed");
+      }
+    },
+    args: () => [this.host],
+  });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.selectedTab = this._getSelectedTab();
+    window.addEventListener("hashchange", this._updateSelectedTab);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener("hashchange", this._updateSelectedTab);
+  }
+
+  firstUpdated() {
+    const params = new URLSearchParams(window.location.search);
+    this.host = params.get("host");
+  }
+
+  _updateSelectedTab() {
+    this.selectedTab = this._getSelectedTab();
```

Comment:
Does this work? I think `this` here refers to `window` when called from `window.addEventListener` -> we should bind `this` to the component in `connectedCallback`, like I do here:

https://github.com/mdn/yari/blob/759eb44b1f1685c8df9955663c4ea3479fd19f19/client/src/lit/play/runner.js#L132:
```js
    this._onMessage = this._onMessage.bind(this);
    window.addEventListener("message", this._onMessage);
```

Reply:
right...